crates/re_build_examples/src/snippets.rs

```diff
@@ -0,0 +1,179 @@
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs::create_dir_all;
+use std::fs::read_dir;
+use std::fs::read_to_string;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Command;
+
+use indicatif::MultiProgress;
+use rayon::prelude::IntoParallelIterator;
+use rayon::prelude::ParallelIterator;
+
+use crate::wait_for_output;
+
+/// Collect code snippets from `docs/snippets` in the repository and run them to produce `.rrd` files.
+#[derive(argh::FromArgs)]
+#[argh(subcommand, name = "snippets")]
+pub struct Snippets {
+    #[argh(positional, description = "directory to output `rrd` files into")]
+    output_dir: PathBuf,
+}
+
+impl Snippets {
+    pub fn run(self) -> anyhow::Result<()> {
+        create_dir_all(&self.output_dir)?;
+
+        let snippets_dir = re_build_tools::cargo_metadata()?
+            .workspace_root
+            .join("docs/snippets");
+
+        println!("Reading config…");
+        let config = read_to_string(snippets_dir.join("snippets.toml"))?;
+        let config: Config = toml::from_str(&config)?;
+
+        println!("Collecting snippets…");
+        let mut snippets = vec![];
+        for snippet in read_dir(snippets_dir.join("all"))? {
+            let snippet = snippet?;
+            let meta = snippet.metadata()?;
+            let path = snippet.path();
+            let name = path.file_stem().and_then(OsStr::to_str).unwrap().to_owned();
+
+            if meta.is_dir() {
+                println!(
+                    "Skipping {}: directories are implicitly opt-out",
+                    path.display()
+                );
+                continue;
+            }
+
+            if !path.extension().is_some_and(|p| p == "py") {
```

Comment:
giving complex expressions a name can greatly improve readability

```suggestion

            let has_python_extension = path.extension().is_some_and(|p| p == "py");
            if !has_python_extension {
```

Reply:
I think this one is already pretty readable, putting it into a variable feels like I'm repeating myself